utils/env_preferences/src/linux.rs

```diff
@@ -2,70 +2,118 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-pub mod linux_locales_prefs {
-    use std::{
-        borrow::Cow,
-        collections::{HashMap, HashSet},
-        ffi::CStr,
-        ptr,
-    };
+pub mod linux_prefs {
+    use std::{borrow::Cow, collections::HashMap, ffi::CStr, ptr};
 
     use libc::{setlocale, LC_ALL, LC_TIME};
+    use std::str::FromStr;
 
-    fn fetch_locale_settings() -> HashMap<String, String> {
+    #[derive(Hash, Eq, PartialEq, Debug)]
+    pub enum LocaleCategory {
+        LcCTYPE,
```

Comment:
issue: these don't have to match the C names. You can give them ergonomic names like `Numeric`.